sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/QuickPulseCoordinator.java

```diff
@@ -67,18 +68,27 @@ public void run() {
 
     @SuppressWarnings("try")
     private long sendData() {
-        dataFetcher.prepareQuickPulseDataForSend(qpsServiceRedirectedEndpoint);
-        QuickPulseHeaderInfo currentQuickPulseHeaderInfo = dataSender.getQuickPulseHeaderInfo();
+        dataSender.setRedirectEndpointPrefix(qpsServiceRedirectedEndpoint);
+        dataFetcher.prepareQuickPulseDataForSend();
 
-        this.handleReceivedHeaders(currentQuickPulseHeaderInfo);
-        collector.setQuickPulseStatus(currentQuickPulseHeaderInfo.getQuickPulseStatus());
-        switch (currentQuickPulseHeaderInfo.getQuickPulseStatus()) {
+        QuickPulseStatus qpStatus = dataSender.getQuickPulseStatus();
+        collector.setQuickPulseStatus(qpStatus);
+        switch (qpStatus) {
             case ERROR:
                 pingMode = true;
+                // Below line is necessary because there is a case where the last valid request is a post
+                // that came 20s before a failing ping, such as a network related error. In this case, if the
+                // network error continues, we would want pings to remain in backoff mode instead of pinging every
+                // 5 seconds. Subtracting the 40s ensures that the ping sender thinks requests have been failing long
+                // enough to stay in a backoff state if the next ping fails.
+                pingSender.resetLastValidRequestTimeNs((long) (dataSender.getLastValidPostRequestTimeNs() - 40 * 1e9));
```

Comment:
(optional)

A bit more explicit in the code:

```suggestion
                long errorDelayInNs= 40 * 1e9;
                pingSender.resetLastValidRequestTimeNs((long) (dataSender.getLastValidPostRequestTimeNs() - 40 * 1e9));
```